qgym/utils/qiskit_utils.py

```diff
@@ -0,0 +1,96 @@
+"""This module contains utility functions for interactions with the qiskit library."""
+
+from __future__ import annotations
+
+from collections.abc import Hashable
+
+import networkx as nx
+import numpy as np
+from numpy.typing import NDArray
+from qiskit import QuantumCircuit
+from qiskit.converters import circuit_to_dag
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler import Layout
+
+
+def get_interaction_graph(circuit: QuantumCircuit | DAGCircuit) -> nx.Graph:
+    """Create and interaction graph from the provided `circuit`.
+
+    Args:
+        circuit: Circuit to produce the interactions from. Both
+            :class:`~qiskit.circuit.QuantumCircuit` and
+            :class:`~qiskit.dagcircuit.DAGCircuit` representations are accepted.
+
+    Returns:
+        Interaction graph of the `circuit`.
+    """
+    dag = _parse_circuit(circuit)
+
+    if dag.multi_qubit_ops():
+        msg = "no 3+ qubit operations are supported"
+        raise ValueError(msg)
+
+    qreg_to_int = _get_qreg_to_int_mapping(dag)
+    interaction_graph: nx.Graph = nx.empty_graph(dag.num_qubits())
+
+    for op_node in dag.op_nodes(include_directives=False):
+        if len(op_node.qargs) == 1:
+            continue
+        qubit1 = qreg_to_int[op_node.qargs[0]]
+        qubit2 = qreg_to_int[op_node.qargs[1]]
+        interaction_graph.add_edge(qubit1, qubit2)
+
+    return interaction_graph
+
+
+def _parse_circuit(circuit: QuantumCircuit | DAGCircuit) -> DAGCircuit:
+    """Create a DAGCircuit from a QuantumCircuit or DAGCircuit."""
+    return circuit if isinstance(circuit, DAGCircuit) else circuit_to_dag(circuit)
+
+
+def _get_qreg_to_int_mapping(
+    circuit: QuantumCircuit | DAGCircuit,
+) -> dict[Hashable, int]:
+    """Create a mapping from the qubits to integer values."""
+    return {qubit: circuit.qubits.index(qubit) for qubit in circuit.qubits}
+
+
+class QiskitMapperWrapper:
+    """Wrap any qiskit mapper (:class:`~qiskit.transpiler.Layout`) such that it becomes
+    compatible with the qgym framework. This class wraps the qiskit mapper, such that it
+    is compatible with the qgym Mapper protocol, which is required for the qgym
+    benchmarking tools.
+    """
+
+    def __init__(self, qiskit_mapper: Layout) -> None:
+        """Init of the :class:`QiskitMapperWrapper`.
+
+        Args:
+            qiskit_mapper: The qiskit mapper (:class:`~qiskit.transpiler.Layout`) to
+                wrap.
```

Comment:
I would consider increasing the line width wrap limit from 90 to 120.

Reply:
Currently we use `black` as our formatter and as they say right on the front page:
[Pro-tip: If you're asking yourself "Do I need to configure anything?" the answer is "No". Black is all about sensible defaults. Applying those defaults will have your code in compliance with many other Black formatted projects.](https://pypi.org/project/black/#:~:text=Pro%2Dtip%3A%20If%20you%27re%20asking%20yourself%20%22Do%20I%20need%20to%20configure%20anything%3F%22%20the%20answer%20is%20%22No%22.%20Black%20is%20all%20about%20sensible%20defaults.%20Applying%20those%20defaults%20will%20have%20your%20code%20in%20compliance%20with%20many%20other%20Black%20formatted%20projects.)

If we at some point move away from `black`, we might change this.

Reply:
Can we tell `black` to use a line width of 120, i.e., is line width configurable within `black`?